### ISSUE TITLE
Bug: rules documentation reference should point to the installed version

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod#268]
### 🧩 Description

See reproduction and expected behaviour

### 🪜 How to reproduce

1. Install a plugin version with a rule that no longer exist
2. Write code which trigger the rule
3. Open the rule reference link
4. Observe

You land in a 404 page

### ✅ Expected behavior

Rule reference link should have point to the tag of the installed version, not `HEAD`

### 💻 System info

```shell
N/A
```

### ✒️ Additional context

N/A
